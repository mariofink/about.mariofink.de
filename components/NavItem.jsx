import Link from "next/link";
import { useRouter } from "next/router";

const NavItem = ({ caption, href }) => {
  const router = useRouter();
  const activeClasses =
    router.asPath === href ? "border-b-4 border-green-500 " : "text-green-500";
  return (
    <Link href={href}>
      <a className={`py-4 px-2 font-semibold ${activeClasses}`}>{caption}</a>
    </Link>
  );
};

export default NavItem;
