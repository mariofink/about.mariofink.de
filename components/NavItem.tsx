import Link from "next/link";
import { useRouter } from "next/router";

interface props {
  caption: string;
  href: string;
}

const NavItem = ({ caption, href }: props) => {
  const router = useRouter();
  const activeClasses =
    router.asPath === href ? "border-b-4 " : "hover:border-b-4 text-green-500";
  return (
    <Link href={href}>
      <a
        className={`transition-all duration-100 ease-in-out py-4 px-2 font-semibold border-green-500 hover:scale-110 ${activeClasses}`}
      >
        {caption}
      </a>
    </Link>
  );
};

export default NavItem;
