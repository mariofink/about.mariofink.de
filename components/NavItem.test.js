import React from "react";
import renderer from "react-test-renderer";
import NavItem from "./NavItem";

/**
 * Needed because the Link component requires next/router
 */
jest.mock("next/dist/client/router", () => require("next-router-mock"));

describe("NavItem", () => {
  it("renders", () => {
    const tree = renderer
      .create(<NavItem caption="my caption" href="/" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
