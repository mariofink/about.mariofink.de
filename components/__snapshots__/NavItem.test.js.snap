// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem renders 1`] = `
<a
  className="transition-all duration-100 ease-in-out py-4 px-2 font-semibold border-emerald-500 hover:scale-110 hover:border-b-4 text-emerald-500"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  my caption
</a>
`;
